apps/server/src/AppService.ts

```
@@ -14,4 +19,16 @@ export class AppService {
   async getBalance(address: string): Promise<BigNumber> {
     return this.ethersRpcProvider.getBalance(address);
   }
+
+  async getAllEventsFromBlockNumber(blockNumber: number): Promise<object[]> {
+    const currentBlockNumber = await this.ethersRpcProvider.getBlockNumber();
+    const contract = new ethers.Contract(
+      this.configService.getOrThrow('contract.address'),
```

This should be sourced in the constructor instead, so if the contract address doesn't exist, then the app can fail start-up (we prefer early exits instead of silent time-bombs in production)